dolphinscheduler-log-server/src/main/java/org/apache/dolphinscheduler/server/log/LoggerRequestProcessor.java

```diff
@@ -113,8 +114,22 @@ public void process(Channel channel, Command command) {
                 List<String> lines = readPartFileContent(rollViewLogPath,
                         rollViewLogRequest.getSkipLineNum(), rollViewLogRequest.getLimit());
                 StringBuilder builder = new StringBuilder();
+                final int MaxResponseLogSize = 65535;
                 for (String line : lines) {
-                    builder.append(line).append("\r\n");
+                    //If a single line of log is exceed max response size, cut off the line
+                    if (line.getBytes(StandardCharsets.UTF_8).length >= MaxResponseLogSize) {
+                        builder.append(line, 0, MaxResponseLogSize)
+                                .append(" [this line' size is exceed ")
+                                .append(MaxResponseLogSize).append(" bytes, so only ")
+                                .append(MaxResponseLogSize).append(" characters are reserved for performance reasons.]")
+                                .append("\r\n");
+                    }else {
+                        builder.append(line).append("\r\n");
+                    }
+
+                    if (builder.toString().getBytes(StandardCharsets.UTF_8).length >= MaxResponseLogSize) {
+                        break;
```

Comment:
I worry this method will be very slow, you already calculate the size of each line, you don't need to calculate the size again.